samtranslator/translator/logical_id_generator.py

```diff
@@ -55,7 +56,8 @@ def get_hash(self, length=HASH_LENGTH):
 
         data_hash = ""
         if self.data_str:
-            data_hash = hashlib.sha1(bytes(self.data_str)).hexdigest()
+            utf_encoded_data_str = str(self.data_str).encode("utf8")
```

Comment:
I think Py2 default encoding is latin1. This may need to be updated to that but unsure.

Reply:
What is the type of `self.data_str`?

* If it's `bytes` (`data_str = b'test'`), then `str(data_str)` gives "b'test'" on python 3
* If it's `unicode` (`data_str = u'snowman: \u2603'`), then `str(data_str)` raises a `UnicodeEncodeError`

I think what you actually want here is:

```
unicode = type(u'')   # or from six import unicode
s = self.data_str
if isinstance(s, unicode):
    s = s.encode('utf-8')
```


Calling `str` here is likely to do bad things -

Reply:
`self.data_str` is a string data - so in py2 it is instance of basestring and in py3 it is unicode. Your solution is correct. We should encode only if it is unicode and skip otherwise.